website/src/pages/DailyPage/components/DailyCharts.tsx

```diff
@@ -0,0 +1,307 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { ChartContainer, ChartTooltip } from "@/components/ui/chart";
+import { Skeleton } from "@/components/ui/skeleton";
+import { MacroData, Workloads } from "@/types";
+import useApiCall from "@/utils/Hook";
+import { secondToMicrosecond } from "@/utils/Utils";
+import { useEffect, useState } from "react";
+import { CartesianGrid, Legend, Line, LineChart, XAxis, YAxis } from "recharts";
+
+export type DailyChartsProps = {
+  benchmarkType: string;
+};
+
+type NumberDataPoint = { x: string; y: number };
+type StringDataPoint = { x: string; y: string };
+
+type ChartDataItem =
+  | { id: string; data: NumberDataPoint[] }
+  | { id: string; data: StringDataPoint[] };
+
+type ChartDatas = {
+  data: ChartDataItem[];
+  title: string;
+  colors: string[];
+};
+
+const chartConfig: { [key: string]: { label: string; color: string } } = {
+  qpsReads: {
+    label: "Reads",
+    color: "hsl(var(--chart-qpsReads))",
+  },
+  qpsWrites: {
+    label: "Writes",
+    color: "hsl(var(--chart-qpsWrites))",
+  },
+  qpsOther: {
+    label: "Other",
+    color: "hsl(var(--chart-qpsOther))",
+  },
+  qpsTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-qpsTotal))",
+  },
+  tps: {
+    label: "TPS",
+    color: "hsl(var(--chart-tps))",
+  },
+  latency: {
+    label: "Latency",
+    color: "hsl(var(--chart-latency))",
+  },
+  cpuTimeTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-cpuTimeTotal))",
+  },
+  cpuTimeVtgate: {
+    label: "Vtgate",
+    color: "hsl(var(--chart-cpuTimeVtgate))",
+  },
+  cpuTimeVttablet: {
+    label: "Vttablet",
+    color: "hsl(var(--chart-cpuTimeVttablet))",
+  },
+  memBytesVtgate: {
+    label: "Vtgate",
+    color: "hsl(var(--chart-memBytesVtgate))",
+  },
+  memBytesVttablet: {
+    label: "Vttablet",
+    color: "hsl(var(--chart-memBytesVttablet))",
+  },
+  memBytesTotal: {
+    label: "Total",
+    color: "hsl(var(--chart-memBytesTotal))",
+  },
+};
+
+export default function DailyCharts(props: DailyChartsProps) {
+  const { benchmarkType } = props;
+  const workloads: Workloads[] = [
+    "OLTP",
+    "OLTP-READONLY",
+    "OLTP-SET",
+    "TPCC",
+    "TPCC_FK",
+    "TPCC_UNSHARDED",
+    "TPCC_FK_UNMANAGED",
+  ];
+
+  const workloadsQuery = workloads.join("&workloads=");
+
+  const {
+    data: dataDaily,
+    error: dailyError,
+    textLoading: dailyTextLoading,
+  } = useApiCall<MacroData>(
+    `${import.meta.env.VITE_API_URL}daily?workloads=${workloadsQuery}`
+  );
+
+  const chartData = dataDaily.map((item) => ({
+    gitRef: item.git_ref.slice(0, 8),
```

Comment:
Let's have a common function that returns the short version of a Git SHA. That way we can call this function everywhere and if someday we want to change from 8 chars to 7, or 6, it will be easy to do.